comp/dogstatsd/listeners/udp_test.go

```diff
@@ -224,14 +224,17 @@ func TestUDPReceive(t *testing.T) {
 	telemetryStore := NewTelemetryStore(nil, deps.Telemetry)
 	packetsTelemetryStore := packets.NewTelemetryStore(nil, deps.Telemetry)
 	s, err := NewUDPListener(packetChannel, newPacketPoolManagerUDP(deps.Config, packetsTelemetryStore), deps.Config, nil, telemetryStore, packetsTelemetryStore)
-	require.NotNil(t, s)
-	assert.Nil(t, err)
-
+	assert.EventuallyWithT(t, func(c *assert.CollectT) {
+		assert.Nil(t, err)
+		assert.NotNil(t, s, "Failed to create UDP listener")
+	}, 2*time.Minute, 2*time.Second)
```

Comment:
Hm, how does this help? Both `s` and `err` are assigned once and would not change their value after the fact. Am I missing something?